src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/Marshal.cs

```diff
@@ -110,7 +110,7 @@ public static int SizeOf<T>(T structure)
             return SizeOfHelper(structure.GetType(), throwIfNotMarshalable: true);
         }
 
-        [RequiresDynamicCode("Marshalling code for the object might not be available. Use the SizeOf<T> overload instead.")]
+        [RequiresDynamicCode("Marshalling code for the object might not be available. Use the 'sizeof' operator if T is blittable or the SizeOf<T> overload instead.")]
```

Comment:
Don't consider it blocking, but I think this is adding mental overhead - replacing with `SizeOf<T>` is straightforward - this sentence now requires extra thinking because it suggest an optimization.

Suggestion to use `sizeof` would be better done by a specialized "this can be written more efficiently" analyzer.

Reply:
> Suggestion to use sizeof would be better done by a specialized "this can be written more efficiently" analyzer.

Similar "this can be written more efficiently for blittable types" suggestions can be made for all other legacy marshalling methods on `Marshal`.